### ISSUE TITLE
Moving from AAD Graph to MS Graph client library

### ISSUE THREAD

**Issue by ritchxu** [microsoftgraph/msgraph-sdk-dotnet#567]
**Is your feature request related to a problem? Please describe.**
We're moving off AAD Graph library as we port codes from targeting .NET Framework to .NET Core. As documented in this guide https://docs.microsoft.com/en-us/graph/migrate-azure-ad-graph-client-libraries, it appears to be easy to migrate from AAD Graph to MS Graph library. But as we tried it out, there seems to be gaps in API supports. Following types are non-existing in MS Graph client library:
```
Microsoft.Azure.ActiveDirectory.GraphClient.Application
Microsoft.Azure.ActiveDirectory.GraphClient.ServicePrincipal
Microsoft.Azure.ActiveDirectory.GraphClient.KeyCredential
Microsoft.Azure.ActiveDirectory.GraphClient.GraphObject
... possibly more
```
**Describe the solution you'd like**
Did I miss any important doc/information or the API gaps do exist as I described?

**Describe alternatives you've considered**
It isn't clear to me what alternatives are available if following assumptions are true:
- AAD Graph library can't be used in applications targeting .NET Core
- We are migrating from .NET Framework to .NET Core
- MS Graph client library presents significant API gaps

**Comment by ritchxu** [comment]
I found this page https://docs.microsoft.com/en-us/graph/migrate-azure-ad-graph-resource-differences where the differences are documented. How could the recommendation be made to move from AAD Graph library to MS Graph library if there are so many key gaps? This leaves us the only option of hand-writing our own client against AAD Graph REST API then?

**Comment by pschaeflein** [comment]
You can plug in an IAuthenticationProvider into the GraphService client. A DelegateAuthenticationProvider will call *your* code for getting a token.

Or, use the Microsoft.Graph.Auth library (https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth) to plug in one of the MSAL providers.  

Example of the auth providers is in the Graph.Community library (https://github.com/microsoftgraph/msgraph-sdk-dotnet-contrib)

**Comment by ritchxu** [comment]
Thanks @pschaeflein, however, the issue isn't about auth in the process of migrating from AAD Graph library to MS Graph library. But rather in MS Graph lacking key APIs we depends on.

**Comment by MIchaelMainer** [comment]
KeyCredential and Application objects are now available in v1.0. ServicePrincipal is available in the beta endpoint, not yet but soon will be available in the beta library. I'm not sure what the GraphObject is but I assume it is the DirectoryObject concept in Microsoft Graph.

**Comment by djj0809** [comment]
Hi @MIchaelMainer, when will ServicePrincipal move to this library from beta library?

**Comment by MIchaelMainer** [comment]
@djj0809 The client library trails the service API implementation. You can expect the v1.0 client library to support ServicePrincipal a bit after it is released in the v1.0. I don't know when it will be released to v1.0.

**Comment by ddyett** [comment]
no action left here.
